pkg/controller/master/setting/cluster_registration_url.go

```diff
@@ -65,3 +71,16 @@ func (h *Handler) registerCluster(setting *harvesterv1.Setting) error {
 		WithSetID("cluster-registration").
 		Apply(objects)
 }
+
+func (h *Handler) cleanupClusterAgent() {
+	// cleanup rancher related resources
+	// ref: https://rancher.com/docs/rancher/v2.6/en/cluster-provisioning/registered-clusters/#registering-a-cluster
+	// ref: https://rancher.com/docs/rancher/v2.6/en/cluster-provisioning/rke-clusters/rancher-agents/
+	if _, err := h.deploymentCache.Get("cattle-system", "cattle-cluster-agent"); err == nil {
+		if err = h.deployments.Delete("cattle-system", "cattle-cluster-agent", nil); err != nil {
+			logrus.Errorf("Can't delete cattle-system/cattle-cluster-agent deployment, err: %+v", err)
+		}
```

Comment:
should we return and do reconciliation instead?

Reply:
Yeah, after thinking again, it's better to do best effort to clean up `cattle-system/cattle-cluster-agent` deployment if we can find it. Updated it. Thank you.